sceval_frontend/src/containers/Hardware.tsx

```diff
@@ -42,15 +42,21 @@ const useLoadHomeDevices = (home: Home | undefined): LoadDevicesState => {
 
   useEffect(() => {
     if (home) {
-      setState({ ...state, isLoading: true });
+      setState((currentState) => {
+          return { ...currentState, isLoading: true }
+      });
       modeAPI
         .getDevices(home.id)
         .then((devices: Device[]): void => {
-          setState({ ...state, devices: devices, isLoading: false });
+          setState((currentState) => {
+              return { ...currentState, devices: devices, isLoading: false }
+          });
         })
         .catch((error: ErrorResponse) => {
           // Unable to get devices
-          setState({ ...state, isLoading: false });
+          setState((currentState) => {
+              return { ...currentState, isLoading: false }
+          });
```

Comment:
There was a missing dependencies `state` here because we use the variable `state`. However, we can't add `state` to the dependencies array because that will cause an infinite loop because we have setState in this useEffect. Therefore, we have to change the way we call setState without references the `state` variable.